src/state/predictions/index.ts

```diff
@@ -154,14 +149,37 @@ export const fetchRounds = createAsyncThunk<{ [key: string]: ReduxNodeRound }, n
         [reduxNodeRound.epoch.toString()]: reduxNodeRound,
       }
     }, {})
+
+    const publicData = {
+      status,
+      currentEpoch,
+      intervalSeconds,
+      minBetAmount,
+      rounds: roundData,
+      ledgers: {},
+      claimableStatuses: {},
+    }
+
+    if (!account) {
+      return publicData
+    }
+
+    const epochs =
+      currentEpoch > PAST_ROUND_COUNT ? range(currentEpoch, currentEpoch - PAST_ROUND_COUNT) : [currentEpoch]
+
+    // Bet data
+    const ledgerResponses = await getLedgerData(account, epochs)
```

Comment:
shold we use `Promise.all` in this case ?

Reply:
No, we wait multicall response inside that function